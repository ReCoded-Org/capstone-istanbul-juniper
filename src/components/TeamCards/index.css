.ourTeam__teamCards {
  background-color: #b4d0c4;
  display: flex;
  flex-direction: row;
  justify-content: space-evenly;
}

.ourTeam__Header {
  font-family: "Salsa", cursive;
  display: flex;
  align-self: center;
  font-size: 2.3em;
}

.ourTeam {
  display: flex;
  flex-direction: column;
}
