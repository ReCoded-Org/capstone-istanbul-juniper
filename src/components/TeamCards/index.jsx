import React, { useState, useEffect } from "react";
import db from "./../../firebaseConfig";
import MemberCard from "./../MemberCard";
import { useTranslation } from "react-i18next";
import { Row } from "antd";
import "./index.css";

const TeamCards = () => {
  const [t] = useTranslation();
  const [dataState, setDataState] = useState([]);

  const fetchData = async () => {
    const res = await db.collection("team").doc("teamInfo").get();
    const data = res.data();
    setDataState(data.info);
  };

  useEffect(() => {
    fetchData();
  }, []);

  return (
    <div className="ourTeam">
      <Row className="ourTeam__Header">{t("about.ourTeam")}</Row>
      <Row className="ourTeam__teamCards">
        {dataState.map((member) => (
          <MemberCard
            key={member.name + "Card"}
            name={t(`about.teamInfo.${member.name}.name`)}
            avatarUrl={member.avatar}
            bio={t(`about.teamInfo.${member.name}.bio`)}
            role={t(`about.teamInfo.${member.name}.role`)}
            linkedinUrl={member.linkedin}
            githubUrl={member.github}
          />
        ))}
      </Row>
    </div>
  );
};

export default TeamCards;
