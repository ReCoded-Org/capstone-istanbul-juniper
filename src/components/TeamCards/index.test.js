import React from "react";
import renderer from "react-test-renderer";
import TeamCards from ".";

it("TeamCards renders correctly", () => {
  const tree = renderer.create(<TeamCards />).toJSON();
  expect(tree).toMatchSnapshot();
});
