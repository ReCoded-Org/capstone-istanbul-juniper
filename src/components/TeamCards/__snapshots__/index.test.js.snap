// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamCards renders correctly 1`] = `
<div
  className="ourTeam"
>
  <div
    className="ant-row ourTeam__Header"
    style={Object {}}
  >
    Our Team
  </div>
  <div
    className="ant-row ourTeam__teamCards"
    style={Object {}}
  />
</div>
`;
